src/core/cluster.py

```diff
@@ -211,7 +252,17 @@ def bootstrap_server(self) -> str:
         if not self.peer_relation:
             return ""
 
-        return ",".join(sorted([f"{broker.host}:{self.port}" for broker in self.brokers]))
+        if self.config.expose_external:  # implicitly checks for k8s in structured_config
+            return self.bootstrap_servers_external
```

Comment:
**praise**: Seems like a sensible solution to keeping this file copy-able / past-able between VM and K8s